pkgs/servers/nosql/mongodb/mongodb.nix

```
@@ -136,12 +136,12 @@ in stdenv.mkDerivation rec {
   hardeningDisable = [ "fortify3" ];
 
   preBuild = ''
-    sconsFlags+=" CC=$CC"
-    sconsFlags+=" CXX=$CXX"
+    appendToVar sconsFlags "CC=$CC"
+    appendToVar sconsFlags "CXX=$CXX"
   '' + lib.optionalString (!stdenv.hostPlatform.isDarwin) ''
-    sconsFlags+=" AR=$AR"
+    appendToVar sconsFlags "AR=$AR"
   '' + lib.optionalString stdenv.hostPlatform.isAarch64 ''
-    sconsFlags+=" CCFLAGS='-march=armv8-a+crc'"
+    appendToVar sconsFlags "CCFLAGS='-march=armv8-a+crc'"
```

`"CCFLAGS='-march=armv8-a+crc'"` doesn't seem right, especially since `scons`'s setup hook takes good care of the flags and doesn't do Bash expansion before execution. In addition, quoting doesn't appear elsewhere in Nixpkgs.

```suggestion
    appendToVar sconsFlags "CCFLAGS=-march=armv8-a+crc"
```

---

As it was an existing issue before this change, we could also address it later.

---

It seems worth cleaning up now unless there’s reason not to.

---

What am now in

---

@Mj-q8 what is going on here?

---

Hello, I got a problem with the processor, entering, and in some cases, I don't understand, I'm new in the subject, I don't understand the status of the text. I want you to explain to me what the problem is

---

@Mj-q8 Please stay on-topic and elaborate on your point.

If you have any questions about this issue, please let us know how it's related. If your question is related to Nix/Nixpkgs/NixOS, consider asking inside the Help category of [NixOS Discourse](https://discourse.nixos.org/c/learn/9). Otherwise, please post it in a forum related to this topic and peruse the forum rules before posting/commenting.

Please remember that most open-source projects and discussion forums are run by volunteers who devote their spare time to helping others. We are not obligated to answer other people's questions. For dedicated support, you can consult the customer service of related products or find a relevant consulting service you trust.